scraper/scrape_server.py

```
@@ -0,0 +1,303 @@
+app = FastAPI(title="careerops-scraper", version="1.0.0")
+
+
+class ScrapeRequest(BaseModel):
+    url: HttpUrl
+
+    @field_validator("url")
+    @classmethod
+    def reject_localhost(cls, v: HttpUrl) -> HttpUrl:
+        host = (urlparse(str(v)).hostname or "").lower()
+        if host in {"localhost", "127.0.0.1", "::1"} or host.startswith("10.") \
+                or host.startswith("172.") or host.startswith("192.168."):
```

Ruff (0.15.13)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)